Finalise the options (#6224)
camptocamp/ngeo: docs/guidelines.md

};
```

## Options

> We have 3 types of configuragle input/output of an Angular component.

Author: Thanks :-)

### Link between component

> The variable that used to share data between component should be provided throw the HTML attributes.

Author: +1

> ### Options that can be represent ad JSON

Author: Done :-)

> All the options of a component that can be represented as JSON should be injected throw one object in the component

Author: +1

> and should start with the prefix `ngeo` or `gmf` and should be describe in one of the `options.js` files with a

Author: +1

type name who is the same as the injected name.

The styles should use the type
[StyleLike](https://camptocamp.github.io/ngeo/master/jsdoc/module-contribs_gmf_src_options.html#~StyleLike)
and be build with the function
[buildStyle](https://camptocamp.github.io/ngeo/master/jsdoc/module-contribs_gmf_src_options.html#.buildStyle).

> For this kind of options we shouldn't porvide default values and we use the standard injection mechanisme, then

Reviewer: ```suggestion
For this kind of options we don't have to provide  a default value. It uses the standard injection mechanism, then
```

Author: +1

> the application should provide a value normally throw the `dynamic.json`.

Reviewer: ```suggestion
the application should provide a value normally through the `dynamic.json`.
```

> But for the value in the option the component shouldn't fail if the argument is missing.

Reviewer: ```suggestion
For the value in the option the component don't fail if an argument is missing.
```

Author: +1

### Options who is a function

> There is no peconisation for this kind of options.

Reviewer: ```suggestion
There is no recommendation for this kind of options.
```

Author: +1
